src/storyblok-auth-api/handle-requests/oauthApiBaseUrl.ts

```diff
@@ -1,20 +1,14 @@
-const isEuSpace = (spaceId: number) => spaceId >= 0 && spaceId < 1000000
-const isUsSpace = (spaceId: number) => spaceId >= 1000000 && spaceId < 2000000
+import { Region } from '../../session'
 
 /**
- * Given a spaceId, returns the API base url for authenticating with oauth
- * @param spaceId
+ * Given a region, returns the API base url for authenticating with oauth
+ * @param region
  */
-export const oauthApiBaseUrl = (spaceId: number) => {
-  if (isEuSpace(spaceId)) {
-    return 'https://app.storyblok.com/oauth'
-  }
-  if (isUsSpace(spaceId)) {
-    return 'https://app.storyblok.com/v1_us/oauth'
-  } else {
-    // TODO type-safe error handling
-    throw new Error(
-      'The spaceId belongs to an unrecognized region. Supported regions are: EU, US. Please upgrade @storyblok/app-extension-auth to a newer version.',
-    )
+export const oauthApiBaseUrl = (region: Region) => {
+  switch (region) {
+    case 'eu':
+      return 'https://app.storyblok.com/oauth'
+    case 'us':
+      return 'https://app.storyblok.com/v1_us/oauth'
   }
```

Comment:
not throwing any more

Reply:
Oh yes, because the `Region` can only assume those two values. If the conversion of `spaceId` to `region` fails, the error is raised there instead.